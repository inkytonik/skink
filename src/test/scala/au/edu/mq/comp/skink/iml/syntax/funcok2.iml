def funcok2(x): (o) {
    o = x;
}

def foo(x : Bool, y : Int): (o1 : Array, o2, o3) {
    o1 = x + y;
    o2 = x;
    o3 = y;
    o1, o2 = bar();
}

def bar(): (o1, o2) {
    o1 = 1;
    o2 = 2;
}
