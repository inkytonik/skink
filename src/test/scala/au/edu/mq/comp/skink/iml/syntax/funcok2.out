Program (
    Vector (
        FunDef (
            "funcok2",
            Vector (Arg ("x")),
            Vector (Arg ("o")),
            Block (
                Vector (
                    Assign (Vector (Var ("o")), Vector (ELoad (Var ("x"))))))),
        FunDef (
            "foo",
            Vector (TypedArg ("x", BoolType ()), TypedArg ("y", IntType ())),
            Vector (TypedArg ("o1", ArrayType ()), Arg ("o2"), Arg ("o3")),
            Block (
                Vector (
                    Assign (
                        Vector (Var ("o1")),
                        Vector (Add (ELoad (Var ("x")), ELoad (Var ("y"))))),
                    Assign (Vector (Var ("o2")), Vector (ELoad (Var ("x")))),
                    Assign (Vector (Var ("o3")), Vector (ELoad (Var ("y")))),
                    Call (Vector (Var ("o1"), Var ("o2")), "bar", Vector ())))),
        FunDef (
            "bar",
            Vector (),
            Vector (Arg ("o1"), Arg ("o2")),
            Block (
                Vector (
                    Assign (Vector (Var ("o1")), Vector (IntLit (1))),
                    Assign (Vector (Var ("o2")), Vector (IntLit (2))))))))
def funcok2(x): (o) {
    o = x;
}

def foo(x: Bool, y: Int): (o1: Array, o2, o3) {
    o1 = x + y;
    o2 = x;
    o3 = y;
    o1, o2 = bar();
}

def bar(): (o1, o2) {
    o1 = 1;
    o2 = 2;
}

